test/pseudo-tty/test-assert-position-indicator.js

```diff
@@ -5,14 +5,25 @@ const assert = require('assert');
 process.env.NODE_DISABLE_COLORS = true;
 process.stderr.columns = 20;
 
+function checkStrings() {
+  assert.strictEqual('123456789ABCDEFGHI', '1!3!5!7!9!BC!!!GHI');
+}
+
 // Confirm that there is no position indicator.
 assert.throws(
-  () => { assert.strictEqual('a'.repeat(30), 'a'.repeat(31)); },
+  () => { checkStrings(); },
   (err) => !err.message.includes('^'),
 );
 
+process.stderr.columns = 80;
+
 // Confirm that there is a position indicator.
 assert.throws(
-  () => { assert.strictEqual('aaaa', 'aaaaa'); },
-  (err) => err.message.includes('^'),
+  () => { checkStrings(); },
+  {
+    message: 'Expected values to be strictly equal:\n' +
+      '\n' +
+      "'123456789ABCDEFGHI' !== '1!3!5!7!9!BC!!!GHI'\n" +
+      '  ^ ^ ^ ^ ^  ^^^    \n',
```

Comment:
I understand the intention behind showing all differences right away, but I believe this makes it more difficult to read than before.
The output is just not easy to map to the input as easy as before by using two lines and a single indicator for the start of the difference. It was intentional not to "overload" the output and to allow easy matching by using two lines. I can actually better read this concrete example with the old implementation because I do not need to go back and forth from left to right.
![image](https://github.com/user-attachments/assets/33b3872d-74cf-4c46-8021-0851b5cf5cd7)

By using two lines, it is also easier to handle longer input with the indicator (by using !== on a single line, we cut the possible characters into less than half).